src/main/java/org/jenkinsci/plugins/gogs/GogsWebHook.java

```diff
@@ -88,128 +94,144 @@ public static String encode(String data, String key) throws Exception {
+        if (gogsSignature == null || gogsSignature.isEmpty()) {
+            gogsSignature = null;
+        }
+
+        // Get queryStringMap from the URI
+        String queryString = checkNotNull(req.getQueryString(), "The queryString in the request is null");
+        Map queryStringMap = checkNotNull(splitQuery(queryString), "Null queryStringMap");
+
+        //Do we have the job name parameter
+        if (!queryStringMap.containsKey("job")) {
+            result.setStatus(404, "Parameter 'job' is missing.");
+            exitWebHook(result, rsp);
+            return;
         }
-        if (body.startsWith("payload=")) {
-          body = body.substring(8);
+        Object jobObject = queryStringMap.get("job");
+        String jobName;
+        if (jobObject == null) {
+            result.setStatus(404, "No value assigned to parameter 'job'");
+            exitWebHook(result, rsp);
+            return;
+        }else{
+            jobName = jobObject.toString();
         }
 
-        JSONObject jsonObject = JSONObject.fromObject(body);
+        // Get the POST stream
+        String body = IOUtils.toString(req.getInputStream(), DEFAULT_CHARSET);
+        if (!body.isEmpty() && req.getRequestURI().contains("/" + URLNAME + "/")) {
+            String contentType = req.getContentType();
+            if (contentType != null && contentType.startsWith("application/x-www-form-urlencoded")) {
+                body = URLDecoder.decode(body, DEFAULT_CHARSET);
+            }
+            if (body.startsWith("payload=")) {
+                body = body.substring(8);
+            }
+
+            JSONObject jsonObject = JSONObject.fromObject(body);
 
-        String jSecret = null;
-        boolean foundJob = false;
+            String jSecret = null;
+            boolean foundJob = false;
 
-        SecurityContext saveCtx = SecurityContextHolder.getContext();
+            SecurityContext saveCtx = SecurityContextHolder.getContext();
 
-        try {
-            Jenkins jenkins = Jenkins.getActiveInstance();
-            ACL acl = jenkins.getACL();
-            acl.impersonate(ACL.SYSTEM);
+            try {
+                Jenkins jenkins = Jenkins.getActiveInstance();
+                ACL acl = jenkins.getACL();
+                acl.impersonate(ACL.SYSTEM);
 
-            Job job = GogsUtils.find(jobName, Job.class);
+                Job job = GogsUtils.find(jobName, Job.class);
 
-            if (job != null) {
-                foundJob = true;
-				/* secret is stored in the properties of Job */
-				final GogsProjectProperty property = (GogsProjectProperty) job.getProperty(GogsProjectProperty.class);
-				if (property != null) { /* only if Gogs secret is defined on the job */
-					jSecret = property.getGogsSecret(); /* Secret provided by Jenkins */
-				}
+                if (job != null) {
+                    foundJob = true;
+                /* secret is stored in the properties of Job */
```

Comment:
You're right @pascalw .
Note that this Pull Request will be closed and replaced by PR #26